pkg/config/node.go

```diff
@@ -307,18 +309,34 @@ func getSortedBackends(kubeconfigPath string) (backends []Backend, err error) {
 	return backends, err
 }
 
-func GetLBConfig(kubeconfigPath string, apiPort, lbPort, statPort uint16, apiVip net.IP) (ApiLBConfig, error) {
+func GetLBConfig(kubeconfigPath string, apiPort, lbPort, statPort uint16, apiVip net.IP, queyrAlsoApiVipOpt ...bool) (ApiLBConfig, error) {
 	config := ApiLBConfig{
 		ApiPort:  apiPort,
 		LbPort:   lbPort,
 		StatPort: statPort,
 	}
+	queyrAlsoApiVip := true
+	if len(queyrAlsoApiVipOpt) > 0 {
+		queyrAlsoApiVip = queyrAlsoApiVipOpt[0]
+	}
+
 	// LB frontend address: IPv6 '::' , IPv4 ''
 	if apiVip.To4() == nil {
 		config.FrontendAddr = "::"
 	}
-
-	backends, err := getSortedBackends(kubeconfigPath)
+	// since:
+	// 1: Keepalived script for Api vip checks HAProxy load balancer status in addition to localhost-kube-apiserver.
+	// 2: Masters nodes details needed for HAProxy config retrieved from kube-apiserver .
+	// to avoid circular dependency we should first read masters details from localhost:kube-apiserver instead of
+	// api-vip:kube-apiserver.
+	backends, err := getSortedBackends(kubeconfigPath, localhostKubeApiServerUrl)
+	if err != nil {
+		log.Infof("An error occurred while trying to read master nodes details from localhost:kube-apiserver: %v", err)
+		if queyrAlsoApiVip {
+			log.Infof("Trying to read master nodes details from api-vip:kube-apiserver")
+			backends, err = getSortedBackends(kubeconfigPath, "")
+		}
+	}
 	if err != nil {
```

Comment:
I find this logic a bit unintuitive because of the way it splits the err check from the call that produces it. Can we just move this into the block with the second getSortedBackends call?